src/mixins/filing-template-mixin.ts

```diff
@@ -770,15 +801,22 @@ export default class FilingTemplateMixin extends DateMixin {
       }
     }
 
-    // store Name Request
-    this.setNameRequest(cloneDeep(
-      filing.correction.nameRequest ||
-      {
+    // store Name Request data
+    if (filing.correction.nameRequest) {
+      this.setNameRequest({
+        // convert to NameRequestIF
+        legalType: filing.correction.nameRequest.legalType,
+        nrNum: filing.correction.nameRequest.nrNumber,
+        legalName: filing.correction.nameRequest.legalName
+      } as any)
+    } else {
+      this.setNameRequest({
+        // convert to NameRequestIF
         legalType: entitySnapshot.businessInfo.legalType,
-        legalName: entitySnapshot.businessInfo.legalName,
-        nrNumber: entitySnapshot.businessInfo.nrNumber
-      }
-    ))
+        nrNum: entitySnapshot.businessInfo.nrNumber,
+        legalName: entitySnapshot.businessInfo.legalName
+      } as any)
```

Comment:
Must cast as "any" because we're not specifying all the properties (which we don't need anyway) and `legalName` isn't a property in the shared interface.